driver-core/src/test/unit/com/mongodb/CustomMatchers.groovy

```diff
@@ -55,25 +58,32 @@ class CustomMatchers {
         if (actual.class.name != expected.class.name) {
             return false
         }
-        getFieldNames(actual.class).findAll { !ignoreNames.contains(it) } .collect {
-            if (nominallyTheSame(it)) {
-                return actual."$it".class == expected."$it".class
-            } else if (actual."$it" != expected."$it") {
-                def (a1, e1) = [actual."$it", expected."$it"]
-                if ([a1, e1].contains(null) && [a1, e1] != nullList) {
+        getFields(actual.class).findAll { !ignoreNames.contains(it.name) } .collect {
+            it.setAccessible(true)
```

Comment:
Since it's now accessing the field directly rather than by property, have to make it accessible